File: typescript/Easy/replace_right_element.ts
class ReplaceRightElement {
  constructor(private readonly arr: number[]) {
    this.arr = arr;
  }

  replaceElement() {
    let rightMax: number = -1;

    for (let i = this.arr.length - 1; i >= 0; --i) {
      let newMax: number = Math.max(rightMax, this.arr[i]);
      this.arr[i] = rightMax;
      rightMax = newMax;
      return this.arr;
    }
  }
}

const replaceObj = new ReplaceRightElement([17, 18, 4, 6, 1]);

console.log(replaceObj.replaceElement());

File: python/Easy/replace_right_element.py
def replace_element(arr):
	right_max = -1

	for i in range(len(arr) - 1, -1, -1):
		new_max = max(right_max, arr[i])
		arr[i] = right_max
		right_max = new_max
	return arr



print(replace_element([17,18,4,6,1]))
